### PR TITLE
gh-120120: Minor improvements in colorsys module

### PR DESCRIPTION
**Pull request by SweetyAngel**
- Minor optimization
- Minor links fixes
- Add support for CMYK color system

* Issue: gh-120120

### CODE REVIEW

File: Lib/test/test_colorsys.py [hunk 1]
```diff
@@ -8,6 +8,8 @@ def frange(start, stop, step):
 
 class ColorsysTest(unittest.TestCase):
 
+
+
```
Comment (nineteendo) [inline Lib/test/test_colorsys.py]: Maybe revert this? 3 empty lines is a bit much.

```suggestion
```

File: Lib/colorsys.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
 """Conversion functions between RGB and other color systems.
 
-This modules provides two functions for each color system ABC:
+This modules provides two functions for each color system ABC (or ABCD):
 
   rgb_to_abc(r, g, b) --> a, b, c
   abc_to_rgb(a, b, c) --> r, g, b
+  
+In case of ABCD:
+
```
Comment (nineteendo) [inline Lib/colorsys.py]: Remove trailing space

```suggestion

In case of ABCD:

```

**Comment by SweetyAngel** [comment]
How to fix it in tests: ```Warning: py:func reference target not found: colorsys.rgb_to_cmyk```?
The function exists.

**Comment by nineteendo** [comment]
You need to document the new functions here: https://github.com/python/cpython/blob/main/Doc/library/colorsys.rst

**Comment by SweetyAngel** [comment]
> You need to document the new functions here: https://github.com/python/cpython/blob/main/Doc/library/colorsys.rst

Thanks

File: Doc/library/colorsys.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@
 The :mod:`colorsys` module defines bidirectional conversions of color values
 between colors expressed in the RGB (Red Green Blue) color space used in
 computer monitors and three other coordinate systems: YIQ, HLS (Hue Lightness
-Saturation) and HSV (Hue Saturation Value).  Coordinates in all of these color
+Saturation), HSV (Hue Saturation Value), CMYK (Cyan Magenta Yellow Key).  Coordinates in all of these color
 spaces are floating point values.  In the YIQ space, the Y coordinate is between
 0 and 1, but the I and Q coordinates can be positive or negative.  In all other
 spaces, the coordinates are all between 0 and 1.
```
Comment (nineteendo) [inline Doc/library/colorsys.rst]: Keep line length short:

```suggestion
Saturation), HSV (Hue Saturation Value), CMYK (Cyan Magenta Yellow Key).
Coordinates in all of these color spaces are floating point values.  In the YIQ
space, the Y coordinate is between 0 and 1, but the I and Q coordinates can be
positive or negative.  In all other spaces, the coordinates are all between 0
and 1.
```

File: Doc/library/colorsys.rst [hunk 4]
```diff
@@ -56,10 +56,22 @@ The :mod:`colorsys` module defines the following functions:
 
    Convert the color from HSV coordinates to RGB coordinates.
 
+.. function:: rgb_to_cmyk(r, g, b)
+
+   Convert the color from RGB coordinates to CMYK coordinates.
+
+.. function:: cmyk_to_rgb(c, m, y, k)
+
+   Convert the color from CMYK coordinates to RGB coordinates.
```
Comment (nineteendo) [inline Doc/library/colorsys.rst]: Add `versionadded`:

```suggestion
   Convert the color from RGB coordinates to CMYK coordinates.

.. versionadded:: 3.14

.. function:: cmyk_to_rgb(c, m, y, k)

   Convert the color from CMYK coordinates to RGB coordinates.

.. versionadded:: 3.14
```

**Comment by gaogaotiantian** [comment]
There was a similar attempt which was denied here - https://github.com/python/cpython/pull/114578 . It would be nice to mention what improvement did you do. Is it coding style only? Because CPython generally do not accept cosmetic changes (changes that have no actual effect except for styling). I believe your `max` / `min` solution will hurt the performance. I'm not sure about match, I would guess it would perform worse as well. So your improvements might not be improvements.

As for CMYK, this is a feature request. It feels nice to have extra stuff in the standard library, but it also increases maintenance effort. That's why we are cautious when we try to add something new into standard library. I'm not saying this should not be in `colorsys`, but this is a module that's old and simple (and kind of in maintenance mode), we probably need to justify that. There are many other color systems in the world, should we add all of them?

**Comment by SweetyAngel** [comment]
> There was a similar attempt which was denied here - #114578 . It would be nice to mention what improvement did you do. Is it coding style only? Because CPython generally do not accept cosmetic changes (changes that have no actual effect except for styling). I believe your `max` / `min` solution will hurt the performance. I'm not sure about match, I would guess it would perform worse as well. So your improvements might not be improvements.

You are right so this should have another name

**Comment by SweetyAngel** [comment]
> As for CMYK, this is a feature request. It feels nice to have extra stuff in the standard library, but it also increases maintenance effort. That's why we are cautious when we try to add something new into standard library. I'm not saying this should not be in `colorsys`, but this is a module that's old and simple (and kind of in maintenance mode), we probably need to justify that. There are many other color systems in the world, should we add all of them?

I know that module is not so useful and I have the idea to mark this module as deprecated and remove it in next python versions because I have not faced this module used anywhere. What do you think about it?

**Comment by nineteendo** [comment]
> I have not faced this module used anywhere

It's used [25.7k times on GitHub](https://github.com/search?q=%2F%28%3F%3Afrom%7Cimport%29+colorsys%2F+language%3APython&type=code).

**Comment by SweetyAngel** [comment]
> > I have not faced this module used anywhere
> 
> It's used [25.7k times on GitHub](https://github.com/search?q=%2F%28%3F%3Afrom%7Cimport%29+colorsys%2F+language%3APython&type=code).

Ok, I ve mistaken but the module is really old and should be removed

**Comment by gaogaotiantian** [comment]
Removing modules from standard library is a big thing and last time we did it it required a PEP - https://peps.python.org/pep-0594/. There's a distinct difference between putting a module in maintenance mode and decrecating it. The latter has a cost and should only be justfied when the maintenance cost is even higher. Like I said, `colorsys` is a pretty simple module and the maintenance cost for now is pretty low. It also does not receive many feature requests from users (users that are really using it and need the new feature). So either deprecating the module, or adding new features to it, need some justfication.

**Pull request closed without merging**